### DISCUSSION TITLE
[Feature Request] Access to the PR Target Branch

### DISCUSSION THREAD

**Discussion by cryptiklemur** [vercel/vercel#3828] [Ideas]
If I create a PR from:

`git@github.com:aequasi/www.discordservers.com.git` `feature/foo` branch

to

`git@github.com:discordservers/www.discordservers.com.git` `master`  branch

I would like to have a build env variable that contains `discordservers/www.discordservers.com` and `master`

Right now, we don't get that

commit message would be nice too

**Comment by coetry** [comment]
Hi @aequasi , can you help me better understand how this workflow would help you? Is it mainly an issue of taking different actions based on the environment you're in?

**Comment by cryptiklemur** [comment]
Pretty much, yes. We run a script to fetch secrets from AWS Secrets Manager during the build, and right now, if we create a PR from `development` into `master`, it builds the app using `development` secrets, and then when it merges, it gets deployed with the development secrets

**Comment by styfle** [comment]
Since [Now 18.0.0](https://github.com/zeit/now/releases/tag/now%4018.0.0) you can manage environment variables with `now env` or on the dashboard.

Does this solve your problem?

**Comment by cryptiklemur** [comment]
Unfortunately not. We have two "production" (as far as ZEIT is concerned) environments: `development` and `master`. They have different environment variables between them however

**Comment by styfle** [comment]
> We have two "production" (as far as ZEIT is concerned) environments

Can you elaborate? Are you referring to two different [branch domains](https://zeit.co/blog/branch-domains)?

**Comment by cryptiklemur** [comment]
Yes,

We run everything using the fork model (upstream repo is the org repo, every developer has their own fork).

The upstream org repo has two branches: `development` (the default branch, to make sure devs are PRing against development instead of master), and `master`. `development` is our staging environment. We run tests against this, and use this to make sure that everything is good before going to production. Not all users of this are developers. `master` is *our* production environment. 

Both of these have `production` URLs set up in ZEIT, to ensure that they are getting the domain (these two both have specific (sub)domains that they need to use) aliased properly after a deploy.

**Comment by paul-vd** [comment]
We kind of have the same issue, so we have two different environments for each project.

so for this example, let's say we have a project named `platform` on `acme` org.

## github setup:
URL: https://github.com/acme/platform/tree/develop
staging: develop branch
production: main branch


## vercel setup:
### Projects:
#### Staging pointing to : https://github.com/acme/platform/tree/develop
- [x]  has it's own preview
- [x]  has it's own production

#### Production pointing to: https://github.com/acme/platform/tree/main
- [ ] should have its own previews (this one is the issue)
- [x] has its own production

### Skip Build Step:
Here we have a script that we add to each project which determines if it should build.

```js
const ABORT_BUILD_CODE = 0;
const CONTINUE_BUILD_CODE = 1;

const continueBuild = () => {
  console.log("🟢 - building");
  process.exit(CONTINUE_BUILD_CODE);
};
const abortBuild = (reason) => {
  console.log("⚪️ - build canceled :", reason);
  process.exit(ABORT_BUILD_CODE);
};

const VERCEL_ENV = process.env.VERCEL_ENV;
const BRANCH = process.env.VERCEL_GIT_COMMIT_REF;
const STAGING = process.env.ENV_QUALIFIER == "stg"; <-- this is our global env identifier

const currentBranch = (allowedBranches) => {
  return allowedBranches.some((branch) => branch.indexOf(BRANCH) > -1);
};

if (VERCEL_ENV !== "preview") {
  return true; // production builds should always continue 😸
}

const isProdBranch = currentBranch(["main", "release/", "hotfix/"]);
const isStagingOnProd = STAGING && isProdBranch; // dont run staging when prod branch 
const isProdOnStaging = !STAGING && !isProdBranch; // dont run prod when not prod branch

if (isStagingOnProd || isProdOnStaging) {
  return abortBuild(
    isStagingOnProd
      ? "staging environment on prod branch"
      : "production environment on staging branch"
  );
} else {
  return continueBuild();
}
```

Now the problem here comes in at `isProdBranch`:
```
const isProdBranch = currentBranch(["main", "release/", "hotfix/"]);
```
because when you do a pull request from `develop` into `main`, the current branch would never be prod until the merge has been done, meaning you would not be able to get previews for your production environment. If we maybe had access to the `target` branch we could add more specific build scripts.

But I'm guessing this is not really possible, because if I understand correctly vercel only reads the branch and not the PR?
